Examples/UIComponentExample/Examples/Complex layout/ComplexLayoutViewController.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  ComplexLayoutViewController.swift
+//  UIComponentExample
+//
+//  Created by y H on 2021/7/21.
+//  Copyright © 2021 Luke Zhao. All rights reserved.
+//
+
+import UIComponent
+import UIKit.UIButton
+
+class ComplexLayoutViewController: ComponentViewController {
+  
+  var showWeather: Bool = false
+  
+  static let originalIds = [Cell.Context(fillColor: colors.randomElement()!, id: "1"),
+                            Cell.Context(fillColor: colors.randomElement()!, id: "2"),
+                            Cell.Context(fillColor: colors.randomElement()!, id: "3"),
+                            Cell.Context(fillColor: colors.randomElement()!, id: "4"),
+                            Cell.Context(fillColor: colors.randomElement()!, id: "5")]
+  static let colors: [UIColor] = [.systemRed, .systemBlue, .systemPink, .systemTeal, .systemGray, .systemFill, .systemGreen, .systemGreen, .systemYellow, .systemPurple, .systemOrange]
+  
+  var ids = originalIds {
+    didSet {
+      reloadComponent()
+    }
+  }
+  
+  var tags: [(String, UIColor)] = [("SwiftUI", .systemRed),
```

Comment:
Can you make it a named tuple `(name: String, color: UIColor)` or a struct? so that we don't have to refer to them as `.0` and `.1`

Reply:
✅